activerecord/lib/active_record/associations.rb

```diff
@@ -73,6 +73,18 @@ def corrections
     end
   end
 
+  class InverseOfAssociationRecursiveError < ActiveRecordError #:nodoc:
+    attr_reader :reflection
+    def initialize(reflection = nil)
+      if reflection
+        @reflection = reflection
+        super("Inverse association for #{reflection.name} (#{reflection.options[:inverse_of].inspect} in #{reflection.class_name}) is recursive.")
+      else
+        super("Inverse association is recursive.")
```

Comment:
Could you add a test for this case?

Reply:
Looking at the code path again, this seems like an impossible scenario. The error is always raised with `self`, and I don't think `self` can ever be `nil` in the context of a reflection (or ever?). I was just copying from `InverseOfAssociationNotFoundError`, but that `else` also seems to be impossible as well. Will open a PR to fix that too if we agree.